### DISCUSSION TITLE
YOLO World has extremely low training metrics on LVIS.

### DISCUSSION THREAD

**Discussion by hongliangchen211-source** [ultralytics/ultralytics#25194] [Q&A]
YOLO World has extremely low training metrics on LVIS.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @hongliangchen211-source, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users, where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many common questions may already be answered.

This is an automated response 🤖, and an Ultralytics engineer will also assist you soon.

For this report about extremely low YOLO-World training metrics on LVIS 🐛, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example), including:
- The exact command or code used for training and validation
- Your `ultralytics` version, Python version, PyTorch version, and hardware details
- The model configuration or checkpoint used
- The LVIS dataset configuration and a description of any dataset modifications
- Complete training logs and validation output
- Representative dataset images with labels, if possible

Please also provide as much information as possible about your custom training setup and verify that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by hongliangchen211-source** [reply]
#!/usr/bin/env python3
"""Minimal YOLO-World training on LVIS 866-seen / 1203-class validation."""

from __future__ import annotations

import sys
from pathlib import Path


# ============================== 只修改这里 ==============================
WEIGHTS = "yolov8s-worldv2.pt"
DATA = "datasets/lvis_ovod/lvis_world_seen866_rare337.yaml"

EPOCHS = 30
BATCH = 16
IMGSZ = 640
DEVICE = 0
WORKERS = 8

LR0 = 5e-5
FREEZE = 10
PROJECT = "lvis_runs"
NAME = "yoloworld_lvis_minimal"
# =======================================================================


ROOT = Path(__file__).resolve().parent
for candidate in (ROOT, *ROOT.parents):
    if (candidate / "ultralytics" / "ultralytics").is_dir():
        ROOT = candidate
        break

ULTRALYTICS_ROOT = ROOT / "ultralytics"
sys.path.insert(0, str(ULTRALYTICS_ROOT))

from ultralytics import YOLOWorld  # noqa: E402
from ultralytics.models.yolo.world.train_lvis_ovd import LvisOVDTrainer  # noqa: E402


def resolve_path(value: str) -> Path:
    path = Path(value)
    return path.resolve() if path.is_absolute() else (ROOT / path).resolve()


def main() -> None:
    weights = resolve_path(WEIGHTS)
    data = resolve_path(DATA)

    if not weights.is_file():
        raise FileNotFoundError(f"找不到预训练权重: {weights}")
    if not data.is_file():
        raise FileNotFoundError(f"找不到LVIS配置: {data}")

    # 直接从 .pt 创建，避免先建 YAML 模型再加载权重造成层不匹配。
    model = YOLOWorld(str(weights))

    model.train(
        trainer=LvisOVDTrainer,
        data=str(data),
        epochs=EPOCHS,
        batch=BATCH,
        imgsz=IMGSZ,
        device=DEVICE,
        workers=WORKERS,
        optimizer="AdamW",
        lr0=LR0,
        lrf=0.1,
        weight_decay=5e-4,
        warmup_epochs=3,
        freeze=FREEZE,
        mosaic=0.0,
        mixup=0.0,
        copy_paste=0.0,
        close_mosaic=0,
        max_det=300,
        conf=0.001,
        iou=0.7,
        amp=True,
        cache=False,
        val=True,
        save=True,
        save_period=1,
        project=str(resolve_path(PROJECT)),
        name=NAME,
        exist_ok=False,
    )


if __name__ == "__main__":
    main()
I conducted the LVIS experiment based on the article that reproduces it.

**Reply by Y-T-G** [reply]
You should use the code that's provided in docs to reproduce:

https://docs.ultralytics.com/models/yolo-world#launch-training-from-scratch

**Reply by hongliangchen211-source** [reply]
However, the download link for the annotated file seems to be broken.

**Reply by Y-T-G** [reply]
Can you post which link is broken? Maybe your internet blocks it

**Reply by hongliangchen211-source** [reply]
I would also like to ask how to reproduce the COCO fine-tuning experiment described in the paper.

**Reply by glenn-jocher** [reply]
For COCO fine-tuning, load the pretrained checkpoint and use the standard `WorldTrainer`, e.g. `YOLOWorld("yolov8s-worldv2.pt").train(data="coco.yaml", epochs=100, imgsz=640)`; to reproduce the paper’s exact result, please specify the table/model variant since its batch size, schedule, and other hyperparameters must match that experiment.

**Comment by rishig47-dev** [comment]
If the YOLO-World training metrics on LVIS are extremely low, I would first verify the evaluation pipeline before changing the model or training recipe.

LVIS has a much larger vocabulary than COCO (1,203 categories), so the class-name/text-embedding setup is important for YOLO-World. The current Ultralytics implementation explicitly handles LVIS class names during validation because YOLO-World normally starts with COCO-style classes.

I would check these points first:

1. Use the LVIS validation/minival split correctly. For YOLO-World training, Ultralytics' WorldTrainer specifically switches to the LVIS `minival` split for final evaluation.

2. Make sure the model is using the LVIS class names. Don't leave the model configured with the default COCO 80 classes when evaluating LVIS. The validator needs to generate text embeddings for the LVIS class names.

3. Check the Ultralytics version. There have been LVIS/YOLO-World validation issues involving class indexing and zero metrics, so reproducing the result with a current Ultralytics version is important before modifying `detect/val.py` manually.

4. Compare against the pretrained checkpoint first.** Run validation on the pretrained YOLO-World checkpoint before training. If the pretrained model also produces near-zero LVIS metrics, the problem is much more likely to be the validation/class mapping pipeline than your training recipe.

5. Don't immediately modify `DetectionValidator.init_metrics()`. If changing the validator makes the code run but produces all-zero metrics, that is a strong indication that the underlying class mapping/evaluation setup is still incorrect rather than the model simply having poor accuracy.

I would first reproduce the official YOLO-World + LVIS workflow using the current Ultralytics implementation, then compare pretrained vs. trained results. Once the validation pipeline is confirmed, you can investigate learning rate, training data, augmentation, and checkpoint quality.

For reference, Ultralytics' current YOLO-World documentation uses LVIS minival as the validation dataset when reproducing YOLO-World results.

**Reply by pderrenger** [reply]
Please share the exact training/validation command, `ultralytics` version, pretrained checkpoint, resulting metrics, and full text logs so we can reproduce this. Current YOLO-World validation sets embeddings from the dataset class names, while `WorldTrainerFromScratch` evaluates LVIS on `minival`; see the [YOLO-World validation reference](https://docs.ultralytics.com/reference/models/yolo/world/val).
